torchmetrics/functional/image/ssim.py

```diff
@@ -85,9 +106,14 @@ def _ssim_update(preds: Tensor, target: Tensor) -> Tuple[Tensor, Tensor]:
             f" Got preds: {preds.dtype} and target: {target.dtype}."
         )
     _check_same_shape(preds, target)
-    if len(preds.shape) != 4:
+    if kernel_dimension == 2 and len(preds.shape) != 4:
+        raise ValueError(
+            "2D kernel is used. Expected `preds` and `target` to have BxCxHxW shape."
+            f" Got preds: {preds.shape} and target: {target.shape}."
+        )
+    if kernel_dimension == 3 and len(preds.shape) != 5:
         raise ValueError(
-            "Expected `preds` and `target` to have BxCxHxW shape."
+            "3D kernel is used. Expected `preds` and `target` to have BxCxHxWxD shape."
```

Comment:
```suggestion
            "3D kernel is used. Expected `preds` and `target` to have BxCxDxHxW shape."
```

AFAIK this should be standard for 3d. Maybe reflect this in the variable names for the 3d kernel generation as well?